src/model_setup.jl

```diff
@@ -1,6 +1,30 @@
-struct ModelSetup{T<:AbstractFloat,Tprog<:AbstractFloat}
+mutable struct PrognosticVars{T<:AbstractFloat}
```

Comment:
is there a reason you changed this to `mutable`?

Reply:
Yeah, without the `mutable` Enzyme fails (something about mixed internal activity types)